src/Preservation/PreservationFactory.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+import '../BaseLevel.sol';
+import './Preservation.sol';
+
+contract PreservationFactory is Level {
+
+  function createInstance(address _player) override public payable returns (address) {
+    _player;
+    LibraryContract _lib1 = new LibraryContract();
+    LibraryContract _lib2 = new LibraryContract();
+    Preservation _preservation = new Preservation(address(_lib1), address(_lib2));
+    return address(_preservation);
+  }
+
+  function validateInstance(address payable _instance, address _player) override public returns (bool) {
+    _player;
+    Preservation instance = Preservation(_instance);
+    return instance.owner() == _player;  
+  }
+
+  receive() external payable {}
```

Comment:
Is this receive needed 

heres the original 

https://github.com/OpenZeppelin/ethernaut/blob/4f80f8afa47793d9133f03b6c247d8b6b0adc673/contracts/contracts/levels/PreservationFactory.sol